rfd/0-job-worker.md

```diff
@@ -0,0 +1,236 @@
+---
+authors: Joshua Rubin (me@jawa.dev)
+state: draft
+---
+
+# RFD 0 - Prototype Job Worker Service
+
+## What
+
+Implement a prototype job worker service that provides an API to run arbitrary Linux processes.
+
+## Details
+
+### Design Approach
+
+Follow the requirements of [the challenge](https://github.com/gravitational/careers/blob/main/challenges/systems/challenge-1.md) as closely as possible and do not unnecessarily expand the scope.
+
+### Proposed API
+
+The gRPC API is defined in `jobworker.proto` and the worker library interface is defined in `worker.go`.
+
+#### Worker Library
+
+The worker library defines a simple interface with methods to start, stop, query status and get the output of a job. The implementation is responsible for starting jobs, keeping track of their output, status and exit codes.
+
+##### cgroups
+
+The library implementation will be configured at instantiation with limits that can be parsed into values for `cpu.max`, `memory.max` and `io.max`. These values will originate from runtime flags provided to the server with defaults that do not implement limits. All processes will be started in a new cgroup regardless of the values of these limits. Note that for expediency, it is assumed that cgroups v2 are in use on the system running the service.
```

Comment:
that makes sense for `cpu.max` and `memory.max`, however for `io.max` it's a little more challenging since we have to determine devices to apply the defaults to. still possible, just wanted to make sure that the work to do so was worthwhile/within the scope of the challenge.

Reply:
updated